services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/attachment/common/AttachmentEntity.kt

```diff
@@ -165,34 +133,27 @@ class ApplicationAttachmentEntity(
 
         other as ApplicationAttachmentEntity
 
-        if (attachmentType != other.attachmentType) return false
-        if (application != other.application) return false
-
-        return true
+        return id == other.id
```

Comment:
Equals and hashcode functions in this class can possibly be improved. I'm not sure if using the superclass for comparing is a good idea. I know that this pr is not to blame. Maybe something like:

```
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is ApplicationAttachmentEntity) return false

        if (id != other.id) return false
        if (attachmentType != other.attachmentType) return false
        if (applicationId != other.applicationId) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id.hashCode()
        result = 31 * result + attachmentType.hashCode()
        result = 31 * result + applicationId.hashCode()
        return result
    }
```

Fields other than id can possible be removed, but I'm not sure.

Reply:
I'm pretty convinced we could get away with not defining these at all. We don't have any use for them ourselves, as long as we stick to using Lists instead of Sets. But I'm not sure how JPA would like that.